src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -376,6 +385,10 @@ private ConnectionCompleteParams GetConnectionCompleteParams(string connectionTy
                     {
                         cancelTupleToCancellationTokenSourceMap.TryRemove(cancelKey, out sourceValue);
                     }
+                    if (source != null)

```

Comment:
I *might* be wrong here, but we could tighten this up by using `source?.Dispose()`